andela/krypton-ah-backend
# #162414175 Add Search Functionality and Custom filtering based on parameters

#### What does this PR do?
Implement custom based search
#### Description of Task to be completed?
Implement Search Functionality and Custom filtering based on parameters
#### How should this be manually tested?
Make a GET request on Postman on the following three routes

localhost:3000/api/v1/articles/search
localhost:3000/api/v1/articles/search/author
localhost:3000/api/v1/articles/search/tag
localhost:3000/api/v1/articles/search/title

Enter any query parameters on the URL and you should get a response based on your search parameters
#### Any background context you want to provide?
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/162414175
#### Screenshots (if appropriate)
<img width="1416" alt="screenshot 2019-02-01 at 2 03 32 pm" src="https://user-images.githubusercontent.com/20207295/52125067-7f239b80-262b-11e9-8c31-52c61634ce5d.png">
<img width="1419" alt="screenshot 2019-02-01 at 2 03 43 pm" src="https://user-images.githubusercontent.com/20207295/52125074-86e34000-262b-11e9-9dc6-49444ecb21cb.png">

<img width="1421" alt="screenshot 2019-02-05 at 9 07 05 am" src="https://user-images.githubusercontent.com/20207295/52260196-92818000-2925-11e9-95db-23a3cebe5eee.png">


#### Questions:
